src/Components/AdvancedSearch/Internal/AdvancedSearchResultDetailsList/AdvancedSearchResultDetailsList.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { useCallback } from 'react';
+import {
+    IAdvancedSearchResultDetailsListProps,
+    IAdvancedSearchResultDetailsListStyleProps,
+    IAdvancedSearchResultDetailsListStyles
+} from './AdvancedSearchResultDetailsList.types';
+import { getStyles } from './AdvancedSearchResultDetailsList.styles';
+import {
+    classNamesFunction,
+    useTheme,
+    styled,
+    DetailsList,
+    DetailsListLayoutMode,
+    IColumn,
+    IDetailsListProps,
+    IconButton,
+    SelectionMode
+} from '@fluentui/react';
+import { IADTTwin } from '../../../../Models/Constants';
+import { useTranslation } from 'react-i18next';
+const getClassNames = classNamesFunction<
+    IAdvancedSearchResultDetailsListStyleProps,
+    IAdvancedSearchResultDetailsListStyles
+>();
+
+const AdvancedSearchResultDetailsList: React.FC<IAdvancedSearchResultDetailsListProps> = ({
+    twins,
+    searchedProperties,
+    onTwinSelection,
+    styles,
+    adapter
+}) => {
+    const { t } = useTranslation();
+    const twinCount = twins.length;
+
+    const classNames = getClassNames(styles, {
+        theme: useTheme()
+    });
+    const columns: IColumn[] = [
+        {
+            key: `twin-id + ${Math.random()}`,
+            name: t('twinId'),
+            fieldName: '$dtId',
+            minWidth: 50,
+            maxWidth: 200,
+            isResizable: true
+        },
+        {
+            key: 'properties',
```

Comment:
why does the above column need a unique key and this one doesn't?

Reply:
because we need the key so we can attach the icon to the column later on the code in line 74